manifests/install.pp

```diff
@@ -15,13 +15,16 @@
 define kmod::install(
   $ensure=present,
   $command='/bin/true',
-  $file='/etc/modprobe.d/modprobe.conf',
+  $file="/etc/modprobe.conf/${title}.conf",
 ) {
-  kmod::generic {"install ${name}":
-    ensure  => $ensure,
-    type    => 'install',
-    module  => $name,
-    command => $command,
-    file    => $file,
+
+  kmod::setting { "kmod::install ${title}":
+    ensure   => $ensure,
+    module   => $title,
```

Comment:
The point (which I think is missing in the PL docs) is that users might want to do:

```puppet
kmod::install { 'My favorite module':
  name => 'foo',
}
```

in which case, you want to manage `/etc/modprobe.conf/foo.conf` really, not `/etc/modprobe.conf/My favorite module.conf`.